js/btcturk.js

```diff
@@ -518,53 +538,132 @@ module.exports = class btcturk extends Exchange {
          * @returns {[[int]]} A list of candles ordered as timestamp, open, high, low, close, volume
          */
         await this.loadMarkets ();
         const market = this.market (symbol);
         const request = {
-            'pair': market['id'],
+            'symbol': market['id'],
+            'resolution': this.timeframes[timeframe],
         };
+        const until = this.safeInteger (params, 'until', this.milliseconds ());
+        if (since !== undefined) {
+            request['from'] = parseInt (since / 1000);
+        }
+        if (until !== undefined) {
+            request['to'] = parseInt (until / 1000);
+        }
+        if (limit === undefined && since === undefined) {
+            limit = 100; // default value
+        }
         if (limit !== undefined) {
-            request['last'] = limit;
+            if (timeframe === '1y') { // difficult with leap years
+                throw new BadRequest (this.id + ' fetchOHLCV () does not accept a limit parameter when timeframe == "1y"');
+            }
+            const seconds = this.options['timeframeToSeconds'][timeframe];
+            const limitSeconds = seconds * (limit - 1);
+            if (since !== undefined) {
+                const to = parseInt (since / 1000) + limitSeconds;
+                request['to'] = Math.min (request['to'], to);
+            } else {
+                request['from'] = parseInt (until / 1000) - limitSeconds;
+            }
         }
-        const response = await this.graphGetOhlcs (this.extend (request, params));
+        const response = await this.graphGetKlinesHistory (this.extend (request, params));
+        //
+        //    {
+        //        "s": "ok",
+        //        "t": [
+        //          1661990400,
+        //          1661990520,
+        //          ...
+        //        ],
+        //        "h": [
+        //          368388.0,
+        //          369090.0,
+        //          ...
+        //        ],
+        //        "o": [
+        //          368388.0,
+        //          368467.0,
+        //          ...
+        //        ],
+        //        "l": [
+        //          368388.0,
+        //          368467.0,
+        //          ...
+        //        ],
+        //        "c": [
+        //          368388.0,
+        //          369090.0,
+        //          ...
+        //        ],
+        //        "v": [
+        //          0.00035208,
+        //          0.2972395,
+        //          ...
+        //        ]
+        //    }
+        //
         return this.parseOHLCVs (response, market, timeframe, since, limit);
     }
 
+    parseOHLCVs (ohlcvs, market = undefined, timeframe = '1m', since = undefined, limit = undefined) {
+        const results = [];
+        const t = this.safeValue (ohlcvs, 't');
+        const h = this.safeValue (ohlcvs, 'h');
+        const o = this.safeValue (ohlcvs, 'o');
+        const l = this.safeValue (ohlcvs, 'l');
```

Comment:
Currently, the build is broken because we have a linting rule that does allow variables like `l` or `o` because it's very easy to confuse them with `1` or `0`, can you fix it?